gatsby-node.js

```diff
@@ -2,75 +2,82 @@ const path = require("path")
-                fieldValue
+  const result = await graphql(
+    `
+      {
+        allMarkdownRemark(
+          sort: { fields: [frontmatter___date], order: DESC }
+          limit: 1000
+        ) {
+          categories: group(field: frontmatter___categories) {
+            fieldValue
+          }
+          authors: group(field: frontmatter___author) {
+            fieldValue
+          }
+          edges {
+            node {
+              fields {
+                slug
               }
-              edges {
-                node {
-                  fields {
-                    slug
-                  }
-                  frontmatter {
-                    title
-                  }
-                }
+              frontmatter {
+                title
               }
             }
           }
-        `
-      ).then(result => {
-        if (result.errors) {
-          console.log(result.errors)
-          reject(result.errors)
         }
+      }
+    `
+  )
+  if (result.errors) {
+    reporter.panicOnBuild(`Error while running Graphql query.`)
+  }
 
-        // Create blog posts pages.
-        const posts = result.data.allMarkdownRemark.edges
+  const blogPost = path.resolve("./src/templates/blog-post.tsx")
+  const catePage = path.resolve("./src/templates/category-template.tsx")
+  const authorPage = path.resolve("./src/templates/author-template.tsx")
 
-        result.data.allMarkdownRemark.group.forEach(cate => {
-          const path = `/Categories/${cate.fieldValue}`
-          createPage({
-            path,
-            component: catePage,
-            context: {
-              cate: cate.fieldValue,
-            },
-          })
-        })
+  const categories = result.data.allMarkdownRemark.categories
+  const posts = result.data.allMarkdownRemark.edges
+  const authors = result.data.allMarkdownRemark.authors
 
-        posts.forEach((post, index) => {
-          const previous =
-            index === posts.length - 1 ? null : posts[index + 1].node
-          const next = index === 0 ? null : posts[index - 1].node
-          const blogPath = `/blogs/${_.kebabCase(post.node.frontmatter.title)}`
+  posts.forEach((post, index) => {
+    const previous = index === posts.length - 1 ? null : posts[index + 1].node
+    const next = index === 0 ? null : posts[index - 1].node
+    const blogPath = `/blogs/${_.kebabCase(post.node.frontmatter.title)}`
+    createPage({
+      path: blogPath,
+      component: blogPost,
+      context: {
+        slug: post.node.fields.slug,
+        previous,
+        next,
+      },
+    })
+  })
 
-          createPage({
-            path: blogPath,
-            component: blogPost,
-            context: {
-              slug: post.node.fields.slug,
-              previous,
-              next,
-            },
-          })
-        })
-      })
-    )
+  categories.forEach(category => {
+    const path = `/Categories/${category.fieldValue}`
```

Comment:
if this is a url i would make it lower case `c`. Also can a category have multiple words? If so we may want to convert it into a slug to be safe, for example `React Native -> react-native`. Something to consider!

Reply:
Yes. It makes sense.